src/realm/object-store/c_api/object.cpp

```diff
@@ -157,6 +157,24 @@ RLM_API const void* _realm_object_get_native_ptr(realm_object_t* obj)
     return static_cast<const Object*>(obj);
 }
 
+RLM_API realm_object_t* realm_object_freeze(const realm_object_t* live_object, const realm_t* frozen_realm)
+{
+    return wrap_err([&]() {
+        const auto realm = *frozen_realm;
```

Comment:
This copies the `realm_t` (so, the `std::shared_ptr` has its refcount incremented).

Use `const auto& realm = ...` instead. :-)